discovery.md

```diff
@@ -691,108 +695,102 @@ Content-Type: application/json
 ```
+200 OK
 Content-Type: application/json
 
 [
-  "id": "{id}"
+  {
+    id: "{id}",
+    epoch: "{epoch}"
+  },
   ...
 ]
 ```
+#### `DELETE /services`
 
-#### `POST /services?import`
-
-This MUST add, or update, the specified Services to the list of Services
-available at this Discovery Endpoint. The Body of the request message MUST
-contain an array of zero or more Service Entries. For convinience to clients,
-the presence of the `id` and `epoch` attributes are OPTIONAL. If `id` is
-present then it MUST be retained as a result of this operation and any existing
-Service with that `id` MUST be replaced. If `epoch` is present then the
-`epoch` value of the resulting Service MUST be larger than the maximum of
-the incoming Service and any existing Service with that same `id`. If either
-attribute is not present then the Discovery Endpoint MUST assign appropriate
-value to the missing attribute.
-
-If the Discovery Endpoint is unable to successully add all of the Services
-in the incoming request then an error MUST be generated and none of the
-specified Services are to be added, or updated, in the Discovery Endpoint.
-
-The Services in the request message MUST be processed in the order in which
-they appear. While any error in the overall process MUST rollback any changes
-made due to the request, the processing of each Service in the request MUST be
-applied as if all previous Services in the request were already completed.
-This means that attribute uniqueness checking, such as on the `name`
-attribute, MUST be done on the new state of all Services as if all previously
-specified Services in the incoming request were already successfully processed.
+This MUST delete all of the Services in the Discovery Endpoint that are
+contained in the request.  If any of those Services cannot be deleted for any
+reason (e.g. an `id` is not given or a Service with the given `id` does not
+exist) then the entire request MUST fail with no effect.  The deletion MAY be a
+soft deletion and MAY have a defined time to live (a.k.a. `ttl`).
 
-For example, if a Discovery Endpoint has a Service called `MyService`, it is
-possible to rename that Service to `YourService` and then to add, or update,
-another Service to use `MyService` as its `name`. But this is only true if
-the rename of the first Service happens first the request message.
+If an `id` is not specified in any of the Services the entire request MUST
+fail.  If no Service can be found that corresponds to a specified `id`, the
+entire request MUST fail.  An `epoch` value MAY be omitted.  If a Service
+corresponding to a specified `id` has a higher recorded `epoch` value than the
+one given the entire request MUST fail.
 
-Likewise, while it might be less than optimal, it is technically possible for
-a request to update a Service multiple times if the same `id` is used more
-than once within a request.
+If other service attributes are included in the request, those SHOULD be
```

Comment:
not MUST?

Reply:
That makes a stronger statement clearly.  I would read the difference as saying that there is no use case where an implementer will want to validate every field as a "is this really the thing you want to delete?" check.  As I think about it a record with the same `id` and `epoch` should always be identical but I am not certain that every use case agrees with me.  Maybe there's a risk to mitigate by switching to MUST?  I'm entirely open to changing this, just trying to understand the reasoning and trade offs.

Reply:
From an interop perspective I'd think we'd want to be clear on what each impl will do w.r.t. finding and accepting the DELETE operation. If one rejects a request while another accepts it based on these extra attributes, then I think we'll have an interop issue.